cloud/filestore/libs/vfs_fuse/fs_impl_data.cpp

```diff
@@ -319,20 +353,33 @@ void TFileSystem::Read(
     request->SetOffset(offset);
     request->SetLength(size);
 
-    Session->ReadData(callContext, std::move(request))
-        .Subscribe([=, ptr = weak_from_this()] (const auto& future) {
-            const auto& response = future.GetValue();
-            if (auto self = ptr.lock(); CheckResponse(self, *callContext, req, response)) {
-                const auto& buffer = response.GetBuffer();
-                ui32 bufferOffset = response.GetBufferOffset();
-                self->ReplyBuf(
-                    *callContext,
-                    response.GetError(),
-                    req,
-                    buffer.data() + bufferOffset,
-                    buffer.size() - bufferOffset);
-            }
-        });
+    auto callback = [=, ptr = weak_from_this()] (const auto& future)
```

Comment:
I think it would be better to refactor code like this keeping the completion code inside subscribe instead of moving it to callback
```c++
    auto future;
    if (Config->GetServerWriteBackCacheEnabled()) {
        future = WriteBackCache->ReadData
    } else {
        future = Session->ReadData
    }

    future.Subscribe(...)
```